[docs] ClickHouse integration

stevhliu opened the pull request:
A first draft for querying Hub datasets with ClickHouse :)

- [x] Link to the blog post once it is published

mariosasko commented on docs/source/clickhouse.md:
```diff
@@ -0,0 +1,98 @@
+# ClickHouse
+
+[ClickHouse](https://clickhouse.com/docs/en/intro) is a fast and efficient column-oriented database for analytical workloads, making it easy to analyze Hub hosted datasets with SQL. To get started quickly, use [`clickhouse-local`](https://clickhouse.com/docs/en/operations/utilities/clickhouse-local) to run SQL queries from the command line and avoid the need to fully install ClickHouse.
```
Nit:

```suggestion
[ClickHouse](https://clickhouse.com/docs/en/intro) is a fast and efficient column-oriented database for analytical workloads, making it easy to analyze Hub-hosted datasets with SQL. To get started quickly, use [`clickhouse-local`](https://clickhouse.com/docs/en/operations/utilities/clickhouse-local) to run SQL queries from the command line and avoid the need to fully install ClickHouse.
```

mariosasko reviewed (commented):
Great job!

julien-c reviewed (commented):
cc @anton-l who's a clickhouse fan IIRC!

gingerwizard commented on docs/source/clickhouse.md:
````diff
@@ -0,0 +1,98 @@
+
+```bash
+./clickhouse local -q "
+    SELECT count() AS c, artists 
````
this endpoint is currently broken, even when the data is present we will need

SETTINGS enable_url_encoding = 0, max_http_get_redirects = 1

gingerwizard commented on docs/source/clickhouse.md:
```diff
@@ -0,0 +1,98 @@
+./clickhouse local -q "
+    SELECT count() AS c,
+        horoscope
+    FROM url(hf('blog_authorship_corpus'))
```
this wont work unfortunately, the paths need to be formulated into a pattern

stevhliu commented:
Thanks for the feedback, I've updated examples to include:

- `SETTINGS enable_url_encoding = 0, max_http_get_redirects = 1`
- simplified the visualization example so it's easier to read
- updated the UDF example so it works now

gingerwizard commented on docs/source/clickhouse.md:
````diff
@@ -0,0 +1,134 @@
+For example, let's create a function to return a list of Parquet files from the [blog_authorship_corpus](https://huggingface.co/datasets/blog_authorship_corpus):
+
+```bash
+./clickhouse local -q "
````
problem with using -q is the UDF wont persist across sessions. So this UDF wont be available in the next invocation. You need to actually use a console i.e.

```bash
./clickhouse local
```

and then from the console do all the commands

gingerwizard commented on docs/source/clickhouse.md:
```diff
@@ -0,0 +1,134 @@
+        FROM url('https://datasets-server.huggingface.co/parquet?dataset=' || dataset, 'JSONAsString')
+    )
+
+    SELECT hf('blog_authorship_corpus') AS paths"
```
hugging_paths

gingerwizard commented on docs/source/clickhouse.md:
```diff
@@ -0,0 +1,134 @@
+    CREATE OR REPLACE FUNCTION hugging_paths AS dataset -> (
+        SELECT arrayMap(x -> (x.1), JSONExtract(json, 'parquet_files', 'Array(Tuple(url String))'))
+        FROM url('https://datasets-server.huggingface.co/parquet?dataset=' || dataset, 'JSONAsString')
+    )
```
semi colon needed

gingerwizard commented on docs/source/clickhouse.md:
````diff
@@ -0,0 +1,134 @@
+Let's make this even easier by creating another function that calls `hugging_paths` and outputs all the files based on the dataset name:
+
+```bash
+./clickhouse local -q" 
````
hugging_paths wont be available for use here - see above about need to start session OR specify hugging_paths function here as well

gingerwizard commented on docs/source/clickhouse.md:
````diff
@@ -0,0 +1,144 @@
+Remember to set `enable_url_encoding` to 0 and `max_https_get_redirects` to 1 to redirect to the path of the Parquet files:
+
+```bash
+SET max_http_get_redirects = 1
````
this wont work if you copy, do as follows:

```SET max_http_get_redirects = 1, enable_url_encoding = 0```

gingerwizard reviewed (approved):
One minor change. If thats fixed im 👍

mariosasko reviewed (approved):
Nice! Some nits. Feel free to merge after addressing them.

severo commented:
Not that the documentation build failed here and on main.

stevhliu commented:
Being investigated [here](https://huggingface.slack.com/archives/C02GLJ5S0E9/p1693589869369289)

mishig25 commented:
Thanks for the patience :) 
Doc build should be working now
